### PR TITLE
Switched to better make_supernode interface

### PR DESCRIPTION
**Pull request by etwest**
This allows us to specify a location for a Supernode to be placed for any for our calls to make_supernode

### CODE REVIEW

File: include/supernode.h [hunk 1]
```diff
@@ -62,18 +62,20 @@ class Supernode {
...
+   * Supernode construtors
    * @param n       the total number of nodes in the graph.
    * @param seed    the (fixed) seed value passed to each supernode.
-   * @return        a pointer to loc, the location of the supernode.
+   * @param loc     (Optional) the memory location to put the supernode.
+   * @return        a pointer to the newly created supernode object
    */
-  static Supernode* makeSupernode(void* loc, uint64_t n, uint64_t seed);
-  static Supernode* makeSupernode(const Supernode& s);
+  static Supernode* makeSupernode(uint64_t n, long seed, void *loc = malloc(bytes_size));
```
Comment (Victor-C-Zhang) [inline include/supernode.h:72]: could this ever be called when `bytes_size` is uninitialized?

Comment (etwest) [reply include/supernode.h:72]: Currently you have to call the configure function first. It's non-trivial to change that.

Comment (etwest) [reply include/supernode.h:72]: So yes

Comment (Victor-C-Zhang) [reply include/supernode.h:72]: not ideal, but i guess we can let it go for now. can you put up an issue for this?

**Review by Victor-C-Zhang — APPROVED** [review]
see comment reply

**Pull request merged**